README.md

```diff
@@ -69,8 +69,30 @@ Parameter | Description | Required | Default
 ***
 
 ### upload
+
+Definitions can be uploaded by one of 2 ways:
+
+* Input file - One definition specified by file path
+* Directory - One to many definitions residing in a specified directory, filtered by an optional regex patten
+
+##### inputFile
+
+When uploading by input file, the following parameters are required: `api`, `version` and `inputFile` are required.
+
+##### directory
+
+Similarly by directory, the following parameters are required: `definitionDirectory`. `definitionFileNameRegex` is optional
+
+API name and version will both be generated by parsing the definition. API name will be the title of the definition with any non word characters replaced by underscores. 
+Version will be pulled directly from the definition.
```

Comment:
Why force them to specify for the single file when we could use the same logic to find out the names from the content of the file?

Reply:
Yup, I see your point

I'd be mindful of changing this as current/previous functionality allows the user to specify API Name. In some cases, this may be desired. Currently the parsing of API name uses underscores which while grand, looks a little funky (if that's the right term).

Reply:
Yeah I get you, you could consider giving all the options and if not specified then it work out the name ect, but if it is specified then use the User defined options